protobuf/source_reflection_test.go

```diff
@@ -99,3 +102,97 @@ func TestReflectionNotRegistered(t *testing.T) {
 	require.Error(t, err)
 	assert.Contains(t, err.Error(), "unknown service grpc.reflection.v1alpha.ServerReflection", "unexpected error")
 }
+
+func TestReflectionRoutingHeaders(t *testing.T) {
+	ln, err := net.Listen("tcp", "127.0.0.1:0")
+	require.NoError(t, err)
+	defer ln.Close()
+
+	s := grpc.NewServer()
+	go s.Serve(ln)
+
+	// Ensure that all streams are closed by the end of the test.
+	defer s.GracefulStop()
+
+	drs := &dummyReflectionServer{}
+	rpb.RegisterServerReflectionServer(s, drs)
```

Comment:
We should register the server before calling. `Serve` :
- https://godoc.org/google.golang.org/grpc#Server.RegisterService

This register call just calls `s.RegisterService`, ie
```
func RegisterServerReflectionServer(s *grpc.Server, srv ...) {
	s.RegisterService(&_ServerReflection_serviceDesc, srv)
}
```

>  This must be called before invoking Serve.

Reply:
Meant to register this before `Serve`, thanks. Fixed.